Fix build failure against jpeg-9 (gcc-4.8)

> [ 26%] Building CXX object sources/graphics/JPGReader.cpp.o
> sources/graphics/JPGReader.cpp: In function 'boolean Shared::Graphics::fill_input_buffer(j_decompress_ptr)':
> sources/graphics/JPGReader.cpp:40:9: error: cannot convert 'bool' to 'boolean' in return
>   return true;
>          ^
> source/shared_lib/sources/graphics/JPGReader.cpp:41:1: warning: control reaches end of non-void function [-Wreturn-type]
>  }
>  ^
> make[2]: *** [sources/graphics/JPGReader.cpp.o] Error 1

Upstream-patch: http://megaglest.svn.sourceforge.net/viewvc/megaglest/trunk/source/shared_lib/sources/graphics/JPGReader.cpp?r1=4082&r2=4095&view=patch

--- source/shared_lib/sources/graphics/JPGReader.cpp
+++ source/shared_lib/sources/graphics/JPGReader.cpp
@@ -37,7 +37,7 @@
 }
 static boolean fill_input_buffer (j_decompress_ptr cinfo) {
 	//it is already filled
-	return true;
+	return boolean(true);
 }
 static void skip_input_data (j_decompress_ptr cinfo, long num_bytes) {
 	if (num_bytes > 0) {
